### PR TITLE
[Merged by Bors] - feat: generalize Polynomial.dvd_derivative_iff and IsCoprime.wronskian_eq_zero_iff

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/Derivative.lean [hunk 1]
```diff
@@ -642,6 +642,20 @@ theorem iterate_derivative_X_sub_pow_self (n : ℕ) (c : R) :
 
 end CommRing
 
+section NoZeroDivisors
+
+variable [Semiring R] [NoZeroDivisors R]
+
+@[simp]
+theorem dvd_derivative_iff  {a : R[X]} : a ∣ derivative a ↔ derivative a = 0 where
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Polynomial/Derivative.lean:650]:
```suggestion
theorem dvd_derivative_iff  {P : R[X]} : P ∣ derivative P ↔ derivative P = 0 where
```
Not very deep, but `a` really looks like a constant :)

File: Mathlib/RingTheory/Polynomial/Wronskian.lean [hunk 2]
```diff
@@ -114,21 +114,12 @@ theorem natDegree_wronskian_lt_add {a b : R[X]} (hw : wronskian a b ≠ 0) :
...
-    exact eq_zero_of_dvd_of_degree_lt h (degree_derivative_lt a_nz)
-  mpr h := by simp [h]
-
 /--
 For coprime polynomials `a` and `b`, their Wronskian is zero
 if and only if their derivatives are zeros.
 -/
-theorem _root_.IsCoprime.wronskian_eq_zero_iff {a b : k[X]} (hc : IsCoprime a b) :
+theorem _root_.IsCoprime.wronskian_eq_zero_iff
+    [NoZeroDivisors R] {a b : R[X]} (hc : IsCoprime a b) :
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Polynomial/Wronskian.lean:122]:
```suggestion
    [NoZeroDivisors R] {P Q : R[X]} (hc : IsCoprime a b) :
```
Maybe I am too traditionalist.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Algebra/Polynomial/Derivative.lean [hunk 3]
```diff
@@ -642,6 +642,20 @@ theorem iterate_derivative_X_sub_pow_self (n : ℕ) (c : R) :
 
 end CommRing
 
+section NoZeroDivisors
+
+variable [Semiring R] [NoZeroDivisors R]
+
+@[simp]
+theorem dvd_derivative_iff  {P : R[X]} : P ∣ derivative P ↔ derivative P = 0 where
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Polynomial/Derivative.lean:650]:
```suggestion
theorem dvd_derivative_iff {P : R[X]} : P ∣ derivative P ↔ derivative P = 0 where
```

File: Mathlib/RingTheory/Polynomial/Wronskian.lean (hunk 2 above)
Comment (Ruben-VandeVelde) [reply Mathlib/RingTheory/Polynomial/Wronskian.lean:122]: I reverted this one because this file consistently uses `a` and `b`

Comment (riccardobrasca) [reply Mathlib/RingTheory/Polynomial/Wronskian.lean:122]: OK, no problem.

**Pull request closed without merging**